### PR TITLE
OpenMC API for input generations

### PR DESCRIPTION
**Pull request by sbradnam**
# Description

Updated input generation to use the OpenMC API functions.

## Type of change

Please select what type of change this is.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New benchmark
    - [ ] Non-breaking change which entirely uses exisiting classes, structure etc
    - [ ] Breaking change which has implemented new/modified classes etc
- [x] New feature 
    - [x] Non-breaking change which adds functionality
    - [ ] Breaking change fix or feature that would cause existing functionality to not work as expected

# Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Existing tests in testrun_test.py updated to  work with new functions. 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] General testing 
    - [x] New and existing unit tests pass locally with my changes
    - [ ] Coverage is >80%

### CODE REVIEW

**Comment by alexvalentine94** [comment]
Currently, this is the best approach we could think of for handling the OpenMC dependency across Windows and Linux. i.e. we set a global flag carried through the code. This assumes that OpenMC will only be used on Linux which is currently true. We may want to consider expanding the CI to cover the cases both with and without the optional dependency and also separating the tests out more between codes will help. 'pytest.mark.skipif' can be useful for putting a condition on whether the test is run or not.

File: jade/__openmc__.py
Comment (alexvalentine94) [inline jade/__openmc__.py:1]: Currently, this is the best approach we could think of for handling the OpenMC dependency across Windows and Linux. i.e. we set a global flag carried through the code. This assumes that OpenMC will only be used on Linux which is currently true. We may want to consider expanding the CI to cover the cases both with and without the optional dependency and also separating the tests out more between codes will help. 'pytest.mark.skipif' can be useful for putting a condition on whether the test is run or not.

**Comment by alexvalentine94** [comment]
Coverage improved. In __openmc__.py the reason for coverage is clear. Updates to openmc.py improve coverage - removal of unneccessary code and modification to typical materials to capture weight fractions.

**Review by dodu94 — APPROVED** [review]
Thanks, sounds good to me

**Pull request merged**